### ISSUE TITLE
adding tests for tables built of multiple components

### ISSUE THREAD

**Issue by jraller** [eddyerburgh/avoriaz#51]
I'd like advice about how best to add tests to this project to test the ability to mount a component who's root template element is one of the valid descendants of `<table>` such as '<thead>`, '<body>`, '<tfoot>`, '<tr>`, '<th>`, '<td>`. 

Is there a better way than adding components to a new folder `test/resources/components/table` and adding the tests to the existing `mount.spec.js`?

The components that caused me to want to create these tests render fine in Vue 2, but when I try to mount a component with a root element of `<tr>` the `.html` and `.text` portions of avoriaz are not behaving as expected. `.html` is returning `<!---->` and `.text` is returning an empty string. Those functions are working as expected for other components. Adding these tests would help to diagnose if this is related in any way to avoriaz, or if the problem originates with another part of my test tooling.

**Comment by jraller** [comment]
I added the tests and it appears that avoriaz is not the problem. If this a useful test I can submit a pull request for it.

```
  it('mounts tables and sub components', () => {
    const wrapper = mount(Table);
    expect(wrapper.html()).to.equal('<table><tbody><tr><td>contents</td></tr></tbody></table>');
    expect(wrapper.find('td').length).to.be.greaterThan(0);
  });
  it('mounts sub components of tables', () => {
    const wrapper = mount(Row);
    expect(wrapper.html()).to.equal('<tr><td>contents</td></tr>');
    expect(wrapper.find('td').length).to.be.greaterThan(0);
  });
```

**Comment by eddyerburgh** [comment]
Hi, yes these are useful tests. I would really appreciate a PR!

**Comment by jraller** [comment]
pull request sent. #52

**Comment by jraller** [comment]
Interestingly enough I found a quirk with the tests in my project. I'm using wallabyjs and 
when testing a component that returns a row of a table I'm able to see that `wrapper.text()` is empty, and `wrapper.html()` is `<!---->` but the following lines return the correct values:

```
const cells = wrapper.find('td');
expect(cells[0].html()).to.equal('<td>a</td>');
expect(cells[1].html()).to.equal('<td>3</td>');
```

I suspect that this may have something to do with my having set up wallabyjs to use browser as the environment. Could it be that different portions of avoriaz behave differently in the presence of a browser regardless if attachToDocument is being used or not? That find looks at a different location than text and html do?

In the tests that I added to this project `expect(wrapper.html()).to.equal('<tr><td>contents</td></tr>');` is working just fine for a component that returns a row of a table.
